base/io.jl

```diff
@@ -538,10 +571,11 @@ function done(itr::EachLine, nada)
     itr.ondone()
     true
 end
-next(itr::EachLine, nada) = (readline(itr.stream), nothing)
+
+next(itr::EachLine, nada) = (readline(itr.stream, itr.chomp), nothing)
 eltype(::Type{EachLine}) = String
 
-readlines(s=STDIN) = collect(eachline(s))
+readlines(s::IO=STDIN, chomp::Bool=false) = collect(eachline(s, chomp))
```

Comment:
As far as I can tell no (there is a separate definition for strings). This also matches documentation.